test/cache/cache-lock.test.ts

```diff
@@ -0,0 +1,276 @@
+  'An adapter with the same name but using a local cache should not attempt to acquire a lock',
+  async (t) => {
+    const config = new AdapterConfig(
+      {},
+      {
+        envDefaultOverrides: {
+          CACHE_TYPE: 'redis',
+          CACHE_REDIS_PORT: 6000,
+        },
+      },
+    )
+    const adapter = new Adapter({
+      name: 'TEST',
+      defaultEndpoint: 'test',
+      config,
+      endpoints: [
+        new AdapterEndpoint({
+          name: 'nowork',
+          transport: new NopTransport(),
+        }),
+      ],
+    })
+
+    const config2 = new AdapterConfig(
+      {},
+      {
+        envDefaultOverrides: {
+          CACHE_TYPE: 'local',
+        },
+      },
+    )
+    const adapter2 = new Adapter({
+      name: 'TEST',
+      defaultEndpoint: 'test',
+      config: config2,
+      endpoints: [
+        new AdapterEndpoint({
+          name: 'nowork',
+          transport: new NopTransport(),
+        }),
+      ],
+    })
+
+    const redisClient = new RedisMock() as unknown as Redis
+    let cache: RedisCache | MockCache = new RedisCache(redisClient) // Fake redis
+
+    const dependencies: Partial<AdapterDependencies> = {
+      cache,
+      redisClient,
+    }
+
+    cache = new MockCache(100) // Fake local cache
+
+    const dependencies2: Partial<AdapterDependencies> = {
+      cache,
+    }
+
+    try {
+      await TestAdapter.start(adapter, t.context, dependencies)
+      await TestAdapter.start(adapter2, t.context, dependencies2)
+
+      t.pass()
+    } catch (error) {
+      t.fail(`The following error should not have been thrown: ${error}`)
+    }
+  },
+)
+
+test.serial(
+  'A lock should automatically extend and prevent another adapter from acquiring a lock with the same key',
+  async (t) => {
+    const config = new AdapterConfig(
+      {},
+      {
+        envDefaultOverrides: {
+          CACHE_TYPE: 'redis',
+          CACHE_REDIS_PORT: 6000,
+        },
+      },
+    )
+    const adapter = new Adapter({
+      name: 'TEST',
+      defaultEndpoint: 'test',
+      config,
+      endpoints: [
+        new AdapterEndpoint({
+          name: 'nowork',
+          transport: new NopTransport(),
+        }),
+      ],
+    })
+
+    const config2 = new AdapterConfig(
+      {},
+      {
+        envDefaultOverrides: {
+          CACHE_TYPE: 'redis',
+          CACHE_REDIS_PORT: 6000,
+        },
+      },
+    )
+    const adapter2 = new Adapter({
+      name: 'TEST',
+      defaultEndpoint: 'test',
+      config: config2,
+      endpoints: [
+        new AdapterEndpoint({
+          name: 'nowork',
+          transport: new NopTransport(),
+        }),
+      ],
+    })
+
+    const redisClient = new RedisMock() as unknown as Redis
+    const cache = new RedisCache(redisClient) // Fake redis
+    const dependencies: Partial<AdapterDependencies> = {
+      cache,
+      redisClient,
+    }
+
+    try {
+      await TestAdapter.start(adapter, t.context, dependencies)
+      await sleep(5000)
```

Comment:
We're not using mocked clocks here so this test will actually take 5s, let's change the lock duration to a much smaller number so the test takes 1-2s max